gluepyter/glue_utils.py

```diff
@@ -67,12 +78,31 @@ def get_advanced_links():
     return advanced_links
 
 
-class ErrorWidget:
-    """Wrapper of a HTML widget for showing error message"""
-
-    def __init__(self, e: Exception, path: str) -> None:
-        value = f"{type(e).__name__} at line {e.__traceback__.tb_lineno} of {path}: {e}"
-        self._widget = HTML(value=value)
-
-    def show(self):
-        display(self._widget)
+def nested_compare(value1, value2):
+    # Compare lists
+    if isinstance(value1, list):
+        if isinstance(value2, list):
+            if len(value1) == len(value2):
+                for v1, v2 in zip(value1, value2):
+                    if not nested_compare(v1, v2):
+                        return False
+                return True
+            else:
+                return False
+        else:
+            return False
+    # Compare dict
+    if isinstance(value1, dict):
+        if isinstance(value2, dict):
+            for k1, v1 in value1.items():
+                if k1 in value2.keys():
+                    if not nested_compare(v1, value2[k1]):
+                        return False
+                else:
+                    return False
+            return True
+        else:
+            return False
+    # Compare immutable
+    else:
+        return value1 == value2
```

Comment:
Suggestion for making the code easier to read (less indentation, no `else` statement)

```suggestion
    if isinstance(value1, list) and isinstance(value2, list):
        if not len(value1) == len(value2):
            return False
        
        for v1, v2 in zip(value1, value2):
            if not nested_compare(v1, v2):
                return False
        
        return True

    # Compare dict
    if isinstance(value1, dict) and isinstance(value2, dict):
        for k1, v1 in value1.items():
            if not k1 in value2.keys():
                return False
            
            if not nested_compare(v1, value2[k1]):
                return False
        
        return True
    
    # Compare immutable
    return value1 == value2
```

Reply:
Thanks @martinRenou